web/src/components/FiltersForm.res

```diff
@@ -0,0 +1,148 @@
+  "lte": {...Filter.make("To date"), kind: Date},
+}
+
+let filtersMap = f => filters->Js.Obj.keys->Belt.Array.map(f)->ignore
+
+// A module to manage filters values
+module Values = {
+  type t = Js.Dict.t<string>
+
+  let empty: t = Js.Dict.empty()
+  let get = (xs: t, name: string) => xs->Js.Dict.get(name)
+  let set = (xs: t, name: string, value: string) => xs->Js.Dict.set(name, value)
+
+  // Helper function to load values from the url
+  let loads = values => {
+    let searchParams = URLSearchParams.current
+    filtersMap(name =>
+      searchParams
+      ->URLSearchParams.get(name)
+      ->Js.Nullable.bind((. value) => values->set(name, value))
+    )
+    Js.log4("Set filters from", windowLocationSearch, "to", values)
+    values
+  }
+
+  // Helper function to dump values to an url
+  let dumps = values => {
+    let searchParams = URLSearchParams.current
+    filtersMap(name =>
+      switch values->get(name) {
+      | None
+      | Some("") =>
+        // When a filter is not set or empty, delete the QS
+        searchParams->URLSearchParams.delete(name)
+      | Some(value) =>
+        // Otherwise set the QS
+        searchParams->URLSearchParams.set(name, value)
+      }
+    )
+    let queryString = searchParams->URLSearchParams.toString
+    Js.log4("Set qs from", values, "to", queryString)
+    queryString
+  }
+
+  let use = () => React.useRef(empty->loads).current
+}
+
+// A component to manage user input
+module Field = {
+  let fieldStyle = ReactDOM.Style.make(~marginTop="5px", ~marginBottom="5px", ())
+  @react.component
+  let make = (~name, ~filter, ~values) => {
+    let (value, setValue) = React.useState(_ =>
+      // Initial value is picked from query string, or filter default
+      values->Js.Dict.get(name)->fromMaybe(filter->Filter.getValue)
+    )
+    let onChange = (v, _) => {
+      // Values is the global filter dictionary
+      values->Js.Dict.set(name, v)
+      // We also store the local field value in a react state
+      setValue(_ => v)
+    }
+    switch filter.kind {
+    | Text =>
+      <MTextInput
+        style={fieldStyle} id={name ++ "-input"} placeholder={filter.description} onChange value
+      />
+    | Date =>
+      <div style={fieldStyle}>
+        <DatePicker id={name ++ "-date"} onChange placeholder={filter.description} />
+      </div>
+    }
+  }
+}
+
+module FieldGroup = {
+  @react.component
+  let make = (~children, ~label) =>
+    <MGridItem> <FormGroup label fieldId={label ++ "fid"}> {children} </FormGroup> </MGridItem>
+}
+
+@react.component
+let make = (~updateFilters: string => unit) => {
+  let (show, setShow) = React.useState(_ => false)
+  // We use a (ref) dictionary to store the current filter values
+  let values = Values.use()
+  let applyFilters = _ => {
+    values->Values.dumps->updateFilters
+  }
+  let toggleProps = {
+    "id": "toggle-button",
+    "aria-label": "Details",
+    "aria-labelledby": "titleId toggle-button",
+    "aria-expanded": !show,
```

Comment:
Unfortunately that doesn't work, the button icon doesn't change on expand, but the example is not working too, see: https://www.patternfly.org/v4/components/card#expandable

This probably needs to be done manually, without the CardExpandableContent component.